ang/civicase/case/details/directives/case-details.directive.js

```diff
@@ -43,19 +43,24 @@
    * @param {object} CaseStatus case status service
    * @param {object} CaseType case type service
    * @param {object} CaseDetailsSummaryBlocks case details summary blocks
+   * @param {object} DetailsCaseTab case details case tab service reference
    */
   function civicaseCaseDetailsController ($location, $sce, $rootScope, $scope,
     $document, allowLinkedCasesTab, BulkActions, CaseDetailsTabs, civicaseCrmApi,
     formatActivity, formatCase, getActivityFeedUrl, getCaseQueryParams, $route,
     $timeout, crmStatus, CasesUtils, PrintMergeCaseAction, ts, ActivityType,
-    CaseStatus, CaseType, CaseDetailsSummaryBlocks) {
+    CaseStatus, CaseType, CaseDetailsSummaryBlocks, DetailsCaseTab) {
+    // Makes the scope available to child directives when they require this parent directive:
+    this.$scope = $scope;
```

Comment:
I understand this part. But could you help me understand, which functions from this scope is needed on the child scope?

Reply:
As discussed, this is used by the custom data directive:
https://github.com/compucorp/uk.co.compucorp.civicase/pull/481/files#diff-5b8b870889d45152e9a29d95c3346629R22

This directive is a child of the case-details. Previously we didn't have a scope for the directive, and this is why it could access the case details methods. But since we needed to add a scope attribute so we can specify the type of data we want to display, the directive lost access to the parent directive's methods and properties.